src/AzureRMAssets.ts

```diff
@@ -116,26 +100,63 @@ export interface VersionRedirect {
 }
 
 /**
- * Metadata for a TLE function.
+ * Metadata for all TLE (Template Language Expression) functions.
+ */
+export class FunctionsMetadata {
+    public constructor(public readonly functionMetadata: FunctionMetadata[]) {
+    }
+
+    public findbyName(functionName: string): FunctionMetadata | undefined {
+        const lowerCasedFunctionName: string = functionName.toLowerCase();
+        for (const functionMetadata of this.functionMetadata) {
+            if (functionMetadata.lowerCaseName === lowerCasedFunctionName) {
+                return functionMetadata;
+            }
+        }
+
+        return undefined;
+    }
+
+    public findByPrefix(functionNamePrefix: string): FunctionMetadata[] {
+        const result: FunctionMetadata[] = [];
+        const lowerCasedPrefix: string = functionNamePrefix.toLowerCase();
+        for (const functionMetadata of this.functionMetadata) {
+            if (functionMetadata.lowerCaseName.startsWith(lowerCasedPrefix)) {
+                result.push(functionMetadata);
+            }
+        }
+
```

Comment:
Similar comment as above - `this.functionMetadata.filter(functionMetadata => functionMetadata.lowerCaseName.startsWith(lowerCasedPrefix) )`